packages/x-data-grid-premium/src/tests/aggregation.DataGridPremium.test.tsx

```diff
@@ -181,7 +186,9 @@ describe('<DataGridPremium /> - Aggregation', () => {
 
         setProps({ columns: [column] });
         fireEvent.doubleClick(cell);
-        expect(cell.querySelector('input')).to.equal(null);
+        waitFor(() => {
```

Comment:
Isn't this an async function?

```suggestion
        async waitFor(() => {
```

_(Just saw this commit pass by in my notifications, I make this comment without context)_

Reply:
Do you mean `await waitFor(...`?
I just tried that, and the test times out 🤷‍♂️

Reply:
> I just tried that, and the test times out 🤷‍♂️

Doesn't that just mean the condition in the `waitFor` is never fulfilled. And neither will it be when called synchronously, the test will just end synchronously while it will keep polling in the background forever (or until the test runner is killed).

Reply:
BTW, this will be caught by the upcoming linter [update](https://github.com/mui/mui-x/pull/14232) through the [`testing-library/await-async-utils`](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/await-async-utils.md) rule.

Reply:
> the test will just end synchronously while it will keep polling in the background forever

Ah, I think you're right!
I'll look into it, thanks for pointing this out!

Reply:
It's gone now 👍🏻